viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -146,30 +154,25 @@ ::grpc::Status SLAMServiceImpl::GetPointCloudMap(
 ::grpc::Status SLAMServiceImpl::GetInternalState(
     ServerContext *context, const GetInternalStateRequest *request,
     ServerWriter<GetInternalStateResponse> *writer) {
-    boost::uuids::uuid uuid = boost::uuids::random_generator()();
-    std::string filename = path_to_map + "/" + "temp_internal_state_" +
-                           boost::uuids::to_string(uuid) + ".pbstream";
-    {
-        std::lock_guard<std::mutex> lk(map_builder_mutex);
-        bool ok = map_builder.SaveMapToFile(true, filename);
-        if (!ok) {
-            std::ostringstream oss;
-            oss << "Failed to save the state as a pbstream.";
-            return grpc::Status(grpc::StatusCode::UNAVAILABLE, oss.str());
-        }
+    viam_carto_get_internal_state_response vcgicr;
+    int status = GetInternalStateC(&vcgicr);
+
+    if (status == 1) {
```

Comment:
When this method is added to CartoFacade, we should be using the error code constants defined here rather than raw integers:
https://github.com/viamrobotics/viam-cartographer/blob/main/viam-cartographer/src/carto_facade/carto_facade.h#L87